src/main/java/com/plotsquared/holoplots/HoloPlotsPlugin.java

```diff
@@ -2,63 +2,47 @@
 
-    @SuppressWarnings("deprecation") // Paper deprecation
     @Override
     public void onEnable() {
-        if (!Bukkit.getPluginManager().getPlugin("PlotSquared").getDescription().getVersion().startsWith("7")) {
-            getLogger().severe("PlotSquared 7.x is required for HoloPlots to work!");
+        final String psVersion = Bukkit.getPluginManager().getPlugin("PlotSquared").getDescription().getVersion();
+        if (!psVersion.startsWith("7") || psVersion.split("\\.")[1].charAt(0) < 1) {
+            getLogger().severe("PlotSquared 7.1.0 or higher is required for HoloPlots to work!");
             getLogger().severe("Please update PlotSquared: https://www.spigotmc.org/resources/77506/");
             getLogger().severe("Disabling HoloPlots...");
             Bukkit.getPluginManager().disablePlugin(this);
             return;
         }
-        if (!Bukkit.getPluginManager().getPlugin("HolographicDisplays").getDescription().getVersion().startsWith("3")) {
-            getLogger().severe("HolographicDisplays 3.x is required for HoloPlots to work!");
-            getLogger().severe("Please update HolographicDisplays: https://dev.bukkit.org/projects/holographic-displays/files");
-            getLogger().severe("Disabling HoloPlots...");
-            Bukkit.getPluginManager().disablePlugin(this);
-            return;
-        }
-        HoloPlotsPlugin.THIS = this;
 
         Configuration.load(new File(getDataFolder(), "settings.yml"), Configuration.class);
         Configuration.save(new File(getDataFolder(), "settings.yml"), Configuration.class);
+        Configuration.deserializeSkullTextures(this.getLogger());
 
-        new PacketListener();
-        HOLO = new PSHoloUtil();
-        // Enable metrics
-        new Metrics(this, BSTATS_ID);
-        new PlotAPI().registerListener(HOLO);
-    }
+        final HoloPlots holoPlots = new DefaultHoloPlots(this, new BukkitHologramProviderResolver(this));
+
+        getLogger().info("Using " + holoPlots.provider().getName() + " as the hologram provider");
 
-    /**
-     * Create a new {@link ItemStack} for the head of a specific player.
-     *
-     * @param owner The {@link UUID} of the player for whose head should be created.
-     * @return The {@link ItemStack} for the player head.
-     */
-    public ItemStack getPlayerSkull(UUID owner) {
-        ItemStack itemStack = new ItemStack(Material.PLAYER_HEAD);
-        SkullMeta meta = (SkullMeta) itemStack.getItemMeta();
-        meta.setOwningPlayer(Bukkit.getOfflinePlayer(owner));
-        itemStack.setItemMeta(meta);
-        return itemStack;
+        new PlotAPI().registerListener(new PlotSquaredListener(holoPlots));
+        getServer().getPluginManager().registerEvents(new ChunkListener(holoPlots), this);
+
+        // Enable metrics
+        new Metrics(this, BSTATS_ID).addCustomChart(new SimplePie(
+                "hologram_provider",
+                () -> holoPlots.provider().getName()
```

Comment:
I'd go with a case differentiation between HD and DH. We don't support anything else and should ignore other data.